### PR TITLE
Adds a JSON parsing library for tasks.json

### PR DESCRIPTION
**Pull request by dan-shaw**
Currently, comments are removed when parsing tasks.json. This PR adds a new dependency that keeps user comments.

### CODE REVIEW

File: Extension/package.json [hunk 1]
```diff
@@ -39,7 +39,31 @@
     "Snippets"
   ],
   "activationEvents": [
-    "*"
+    "onLanguage:cpp",
```
Comment (sean-mcmanus) [inline Extension/package.json]: This activationEvent change should be reverted (manually is fine). It's an artifact of the fact that we modify the package.json on runtime, i.e. the normal procedure to update our package.json would be to undo this change first, apply the change, stage/commit the change, then Launch extension (which would change the activationEvents again). The other alternative is to copy the Extension folder to a different location to use for debugging.

**Comment by sean-mcmanus** [comment]
@Colengms Was there a GitHub issue tracking the comment removal issue?

**Comment by Colengms** [comment]
> @Colengms Was there a GitHub issue tracking the comment removal issue?

@sean-mcmanus Are you referring to the issue when editing `c_cpp_properties.json` via the Configuration UI?  I dont think so.

@dan-shaw Can this also be extended to replace existing uses of `jsonc-parser`?  It would be nice to depend on only 1 jsonc parser.

**Review by sean-mcmanus — APPROVED** [review]

**Review by sean-mcmanus — CHANGES_REQUESTED** [review]
I don't see anything fixed by this PR. When I have a tasks.json or c_cpp_properties.json and then trigger an action that causes one of those to be editted by our code, then the comments get overwritten still. Maybe some extra code needs to be added preserver the comments?

**Comment by sean-mcmanus** [comment]
> I don't see anything fixed by this PR. When I have a tasks.json or c_cpp_properties.json and then trigger an action that causes one of those to be editted by our code, then the comments get overwritten still. Maybe some extra code needs to be added preserver the comments?

Oh, I had the wrong branch checked out.

**Comment by sean-mcmanus** [comment]
Closing this PR was accidental, not sure how that happened.

File: Extension/src/common.ts [hunk 2]
```diff
@@ -123,7 +123,7 @@ export async function ensureBuildTaskExists(taskName: string): Promise<void> {
         throw new Error("Failed to get tasksJsonPath in ensureBuildTaskExists()");
     }
 
-    await writeFileText(tasksJsonPath, JSON.stringify(rawTasksJson, null, settings.editorTabSize));
```
Comment (sean-mcmanus) [inline Extension/src/common.ts:126]: Text "// TODO: It's dangerous to overwrite this file. We could be wiping out comments." above should be removed.

File: Extension/src/LanguageServer/configurations.ts [hunk 3]
```diff
@@ -16,7 +16,7 @@ import { getCustomConfigProviders } from './customProviders';
 import { SettingsPanel } from './settingsPanel';
 import * as os from 'os';
 import escapeStringRegExp = require('escape-string-regexp');
-import * as jsonc from 'jsonc-parser';
+import * as jsonc from 'comment-json';
```
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/configurations.ts:19]: This file has to 2 references to JSON.stringify that should be changed to jsonc.stringify.

**Review by sean-mcmanus — CHANGES_REQUESTED** [review]

**Comment by elahehrashedi** [comment]
> @Colengms Was there a GitHub issue tracking the comment removal issue?
This issue happens in Vs Code as well: https://github.com/microsoft/vscode/issues/37138
that's good if we fix it here to preserve comments. However, if the user runs some commands such as "Tasks -> Configure Default Build Task" the comments will be removed by Vs Code.

**Comment by sean-mcmanus** [comment]
@dan-shaw This PR seems close to getting checked in and seems to just need 3 minor changes I mentioned in my previous comments. Did you plan to make those changes or did you want us to do that?

**Comment by dan-shaw** [comment]
@sean-mcmanus Sorry, just fixed it.

**Review by sean-mcmanus — APPROVED** [review]

**Comment by elahehrashedi** [comment]
C/C++ extension will keep the comments while modifying tasks.json file. However, Vs Code doesn't. 
I opened an issue on Vs Code to follow up on this issue: https://github.com/microsoft/vscode/issues/105885

**Pull request merged**